### ISSUE TITLE
Improve test_problems.py

### ISSUE THREAD

**Issue by vishal-ph** [entropicalabs/openqaoa#226]
### Issue Description
- The file is too long and contains tests for all problem classes. We need to break it down as separate files testing each problem class for ease of maintainability.
- The tests contain some unused code and non consistent coding standards that need to be fixed.

**Comment by jiggychauhi** [comment]
Hey, this sound like something a newbie like me can do. Please assign me this issue.

**Comment by devilkiller-ag** [comment]
Hi, Can I work on this issue?

**Comment by vishal-ph** [comment]
Hello, thanks for your interest in contributing to OpenQAOA! 
We encourage you to collaborate on this issue, otherwise, feel free to make an attempt individually and raise a PR to `openqaoa/dev` branch.

**Comment by devilkiller-ag** [comment]
@vishal-ph , I have moved all the tests into different files. Kindly heck my PR if it's ready for merge with dev branch or not.

**Comment by devilkiller-ag** [comment]
@vishal-ph  any updates?

**Comment by vishal-ph** [comment]
@devilkiller-ag, I have added my comments on the PR itself

**Comment by Newtech66** [comment]
@vishal-ph The path is `tests/test_problems.py` right, not `src/openqaoa-core/tests/test_problems.py`?
So, we want it to be broken into `tests/test_problems/test_*.py`? (where * is a problem class)

**Comment by devilkiller-ag** [comment]
@vishal-ph the hyphen in folder name 'openqaoa-core' is causing an import problem. Can I rename it?

**Comment by vishal-ph** [comment]
@Newtech66, no, the path is correct. You will need to look at the `dev` branch. Note the comment at the end of issue description

**Comment by vishal-ph** [comment]
@devilkiller-ag, yes, you are right about the import name conventions. However, openqaoa-core is just the name of the module, the import statement still works with `openqaoa`. Similarly for the remaining plug-ins, they are imported with underscores in the names. For instance, `openqaoa-qiskit` will be imported as `openqaoa_qiskit`.

**Comment by Newtech66** [comment]
@vishal-ph Ok, so all the new files should live in the same path as `test_problems.py`? Not a subfolder `./test_problems/` right?

**Comment by vishal-ph** [comment]
Yes, that's correct. Although eventually, we would want to put all the problem class tests in the `test_problems` folder as you suggested. This would then require updating the test workflows, so that can be done later.
